src/match_taxa_gbif_backbone.Rmd

```diff
@@ -0,0 +1,471 @@
+---
+title: "Match taxa with GBIF Taxonomy Backbone"
+author:
+  - Damiano Oldoni
+  - Lien Reyserhove
+  - Peter Desmet
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: true
+    toc_depth: 3
+    toc_float: true
+    number_sections: true
+    theme: yeti
+    df_print: paged
+knit: (function(input_file, encoding) { rmarkdown::render(input_file, encoding = encoding, output_file = paste0("../docs/",sub(".Rmd", ".html", basename(input_file))))})
+---
+
+# Setup
+
+## Load libraries
+
+```{r load_libs}
+library(odbc)       # To work with database
+library(tidyverse)  # To do data science
+library(here)       # To work with paths
+library(inborutils) # To match taxa (rgbif called in background)
+```
+
+## Load access informations
+
+Retrieve access informations from configuration file:
+
+```{r get_access_infos}
+ibge_bim <- config::get("ibge_bim")
+```
+
+# Read taxa
+
+Read taxa from `./data/interim/taxa.tsv`:
+
+```{r read_taxa_from_tsv_file}
+taxa <- read_tsv(here("data", "interim", "taxa.tsv"),
+                 col_types = cols(
+                   .default = col_character(),
+                   id = col_double(),
+                   parentid = col_double(),
+                   media = col_logical(),
+                   speciesbeparentid = col_double(),
+                   bruenvi_created = col_date(format = "%Y-%m-%d"),
+                   bruenvi_created = col_date(format = "%Y-%m-%d"),
+                   kingdom_id = col_double()
+                   )
+)
+```
+
+# Match taxa to GBIF Taxonomy Backbone
+
+## Match by `acceptedname` + `scientificnameauthorship`
+
+We try to match by using `acceptedname` + `scientificnameauthorship`.
+First, we prepare `taxa` for matching. In this step:
+
+1. We create a new column, `name_for_gbif`, where we paste `acceptedname` and `scientificnameauthorship`, if present.
+2. Recode the taxon ranks from `taxonranken` in a new column, `rank_for_gbif`, containing the correct GBIF terms
+
+```{r prepare_taxa_for_matching_backbone}
+taxa <-
+  taxa %>%
+  mutate(name_for_gbif = if_else(
+    !is.na(scientificnameauthorship),
+    paste(acceptedname, scientificnameauthorship),
+    acceptedname
+  )) %>%
+  mutate(rank_for_gbif = recode(taxonranken,
+    "forma" = "form",
+    "subforma" = "subform",
+    "informal group" = NA_character_
+  ))
+```
+
+We match the names in `name_for_gbif` with GBIF Backbone. We also use the columns containing kingdom information, `bim_kingdom`, and rank, `rank_for_gbif`, and we require a strict match, `strict = TRUE` to avoid fuzzy match with higher classification. **Note: This step takes long**:
+
+```{r get_name_gbif}
+taxa_gbif <-
+  taxa %>%
+  gbif_species_name_match(name = "name_for_gbif",
+                          kingdom = "bim_kingdom", 
+                          rank = "rank_for_gbif",
+                          strict = TRUE)
+```
+
+Some columns containing GBIF data have been added. Preview:
```

Comment:
Is it an option to add the prefix `gbif_` to the newly added columns? Would make it more clear which columns have been added exactly.

Reply:
Good idea! Thanks. I did it: https://github.com/inbo/ibge-bim-species/pull/10/commits/86b051334f10e27e65e17482d2ad9bb383bf9005